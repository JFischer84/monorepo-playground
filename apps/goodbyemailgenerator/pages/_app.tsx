import { AppProps } from 'next/app';
import Head from 'next/head';
import { ThemeProvider } from '@mui/material/styles';
import CssBaseline from '@mui/material/CssBaseline';
import { CacheProvider, EmotionCache } from '@emotion/react';
import theme from '../config/theme';
import createEmotionCache from '../config/createEmotionCache';

interface MyAppProps extends AppProps {
  emotionCache?: EmotionCache;
}

const clientSideEmotionCache = createEmotionCache();

function CustomApp({ Component, emotionCache = clientSideEmotionCache, pageProps }: MyAppProps) {
  return (
    <CacheProvider value={emotionCache}>
      <Head>
        <title>Welcome to goodbyemailgenerator!</title>
      </Head>
      <ThemeProvider theme={theme}>
        <CssBaseline />
        <main className="app">
          <Component {...pageProps} />
        </main>
      </ThemeProvider>
    </CacheProvider>

  );
}

export default CustomApp;
