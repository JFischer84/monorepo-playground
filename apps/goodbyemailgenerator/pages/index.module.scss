.mainContainer {
  display: flex;
  flex-direction: row;
}

.mainCard {
  min-width: 70%;
  margin: 2em;
}
