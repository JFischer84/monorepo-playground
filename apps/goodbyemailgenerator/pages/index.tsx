import styles from './index.module.scss';
import {Button, Card, CardActions, CardContent, FormControlLabel, FormGroup, Switch, Typography} from '@mui/material';
import CasinoIcon from '@mui/icons-material/Casino';
import React, {useState} from 'react';
import {subjects, animalMetaphors, sportMetaphors, intros, goodByePhrases, typicalPhrases, salutations} from './phrases';

export function Index() {

  const [subject, setSubject] = useState('Betreff');
  const [subjectChecked, setSubjectChecked] = useState(false);
  const [animalMetaphor, setAnimalMetaphor] = useState('');
  const [amChecked, setAmChecked] = useState(false);
  const [sportMetaphor, setSportMetaphor] = useState('');
  const [smChecked, setSmChecked] = useState(false);
  const [intro, setIntro] = useState('');
  const [introChecked, setIntroChecked] = useState(false);
  const [goodByePhrase, setGoodByePhrase] = useState('');
  const [gBPChecked, setGBPChecked] = useState(false);
  const [typicalPhrase, setTypicalPhrase] = useState('');
  const [tPChecked, setTPChecked] = useState(false);
  const [salutation, setSalutation] = useState('');
  const [salutationChecked, setSalutationChecked] = useState(false);

  const getRandomNumber = (max: number): number => {
    return Math.floor(Math.random() * max);
  };

  const randomize = (isChecked: boolean, phrases: string[], setter: Function): void => {
    if (isChecked) {
      setter(phrases[getRandomNumber(phrases.length - 1)]);
    }
  };

  const copyText = (): void => {
    navigator.clipboard.writeText(`${salutation}\n${intro}\n${animalMetaphor}\n${sportMetaphor}\n${typicalPhrase}\n${goodByePhrase}\n`);
  };

  const handleSubjectChange = (event: React.ChangeEvent<HTMLInputElement>) => {
    console.log('jojojo');
    if (!subjectChecked) {
      setSubject(subjects[getRandomNumber(subjects.length - 1)]);
    } else {
      setSubject('Betreff');
    }
    setSubjectChecked(event.target.checked);
  };

  const handleSalutationChange = (event: React.ChangeEvent<HTMLInputElement>) => {
    if (!salutationChecked) {
      setSalutation(salutations[getRandomNumber(salutations.length - 1)]);
    } else {
      setSalutation('');
    }
    setSalutationChecked(event.target.checked);
  };

  const handleIntroChange = (event: React.ChangeEvent<HTMLInputElement>) => {
    if (!introChecked) {
      setIntro(intros[getRandomNumber(intros.length - 1)]);
    } else {
      setIntro('');
    }
    setIntroChecked(event.target.checked);
  };

  const handleAmChange = (event: React.ChangeEvent<HTMLInputElement>) => {
    if (!amChecked) {
      setAnimalMetaphor(animalMetaphors[getRandomNumber(animalMetaphors.length - 1)]);
    } else {
      setAnimalMetaphor('');
    }
    setAmChecked(event.target.checked);
  };

  const handleSmChange = (event: React.ChangeEvent<HTMLInputElement>) => {
    if (!smChecked) {
      setSportMetaphor(sportMetaphors[getRandomNumber(sportMetaphors.length - 1)]);
    } else {
      setSportMetaphor('');
    }
    setSmChecked(event.target.checked);
  };

  const handleTPChange = (event: React.ChangeEvent<HTMLInputElement>) => {
    if (!tPChecked) {
      setTypicalPhrase(typicalPhrases[getRandomNumber(typicalPhrases.length - 1)]);
    } else {
      setTypicalPhrase('');
    }
    setTPChecked(event.target.checked);
  };

  const handleGBPChange = (event: React.ChangeEvent<HTMLInputElement>) => {
    if (!gBPChecked) {
      setGoodByePhrase(goodByePhrases[getRandomNumber(goodByePhrases.length - 1)]);
    } else {
      setGoodByePhrase('');
    }
    setGBPChecked(event.target.checked);
  };

  return (
    <>
      <h1>Good Bye Mail Generator!</h1>
      <div className={styles.mainContainer} >
      <Card className={styles.mainCard} variant="outlined">
        <CardContent>
          <Typography color="text.secondary" gutterBottom>
            {subject}
          </Typography>
          <Typography variant="body2">
            {salutation}
            <br />
            {intro}
            <br />
            {animalMetaphor}
            <br />
            {sportMetaphor}
            <br />
            {typicalPhrase}
            <br />
            {goodByePhrase}
          </Typography>
        </CardContent>
        <CardActions>
          <Button onClick={() => copyText()} size="small">Copy</Button>
        </CardActions>
      </Card>
        <div>
          <FormGroup>
          <FormControlLabel
            control={
            <Switch checked={subjectChecked} onChange={handleSubjectChange} inputProps={{ 'aria-label': 'controlled' }} />
            }
            label="Setze Betreff"
          />
            <Button className={styles.randomizeButton} size="small" variant="contained" onClick={() => randomize(subjectChecked, subjects, setSubject)}><CasinoIcon /></Button>

            <FormControlLabel
              control={
                <Switch checked={salutationChecked} onChange={handleSalutationChange} inputProps={{ 'aria-label': 'controlled' }} />
              }
              label="Setze Anrede"
            />
            <Button className={styles.randomizeButton} size="small" variant="contained" onClick={() => randomize(salutationChecked, salutations, setSalutation)}><CasinoIcon /></Button>
            <FormControlLabel
              control={
                <Switch checked={introChecked} onChange={handleIntroChange} inputProps={{ 'aria-label': 'controlled' }} />
              }
              label="Setze Eingangsphrase"
            />
            <Button className={styles.randomizeButton} size="small" variant="contained" onClick={() => randomize(introChecked, intros, setIntro)}><CasinoIcon /></Button>

            <FormControlLabel
              control={
                <Switch checked={amChecked} onChange={handleAmChange} inputProps={{ 'aria-label': 'controlled' }} />
              }
              label="Setze Tier-Metapher"
            />
            <Button className={styles.randomizeButton} size="small" variant="contained" onClick={() => randomize(amChecked, animalMetaphors, setAnimalMetaphor)}><CasinoIcon /></Button>

            <FormControlLabel
              control={
                <Switch checked={smChecked} onChange={handleSmChange} inputProps={{ 'aria-label': 'controlled' }} />
              }
              label="Setze Sport-Metapher"
            />
            <Button className={styles.randomizeButton} size="small" variant="contained" onClick={() => randomize(smChecked, sportMetaphors, setSportMetaphor)}><CasinoIcon /></Button>

            <FormControlLabel
              control={
                <Switch checked={tPChecked} onChange={handleTPChange} inputProps={{ 'aria-label': 'controlled' }} />
              }
              label="Setze typische Phrase"
            />
            <Button className={styles.randomizeButton} size="small" variant="contained" onClick={() => randomize(tPChecked, typicalPhrases, setTypicalPhrase)}><CasinoIcon /></Button>

            <FormControlLabel
              control={
                <Switch checked={gBPChecked} onChange={handleGBPChange} inputProps={{ 'aria-label': 'controlled' }} />
              }
              label="Setze Abschiedsfloskel"
            />
            <Button className={styles.randomizeButton} size="small" variant="contained" onClick={() => randomize(gBPChecked, goodByePhrases, setGoodByePhrase)}><CasinoIcon /></Button>
        </FormGroup>
        </div>
      </div>
    </>

  );
}

export default Index;
