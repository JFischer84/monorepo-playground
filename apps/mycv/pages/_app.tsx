import React from 'react';
import './app.scss';
import {AppProps} from 'next/app';
import Head from 'next/head';

const CustomApp = ({Component, pageProps}: AppProps) => (
  <>
    <Head>
      <title>Welcome to mycv!</title>
    </Head>
    <main className="app">
      {/* eslint-disable-next-line react/jsx-props-no-spreading */}
      <Component {...pageProps} />
    </main>
  </>
);

export default CustomApp;
