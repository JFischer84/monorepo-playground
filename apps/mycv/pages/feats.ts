export default [
  'Jest',
  'Enzyme',
  'JUnit',
  'Mockito',
  'Elasticsearch',
  'Git',
  'Scrum'
];
