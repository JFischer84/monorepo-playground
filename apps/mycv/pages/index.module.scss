.page {
  .uperContainer {
    display: flex;
    align-items: center;
    flex-direction: column;
    justify-content: space-between;
    margin-bottom: 36px;

    .profilePicture {
      border-radius: 50%;
    }

    .imageContainer {
      margin-bottom: 24px;
    }

    .factSection {
      margin-top: 4px;
      border: 2px solid #000000B2;
      border-radius: 2%;
      padding: 6px;
      width: 100%;
      min-width: 320px;
      display: flex;
      align-items: center;
      flex-direction: column;

     /* @media only screen and (min-width: 601px) {
        align-items: baseline;
        flex-direction: row;
        justify-content: space-between;
      }*/


      .stat {
        display: block;
        font-weight: bold;
        margin-bottom: 0;
      }
    }
  }
}
