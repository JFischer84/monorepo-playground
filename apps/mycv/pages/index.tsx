import React from 'react';
import Image from 'next/image';
import styles from './index.module.scss';
import {ProgressBar} from '../app/ProgressBar';
import {Fact} from '../app/Fact';

export function Index() {
  const openLink = (link: string) => {
    window.open(link);
  };
  return (
    <div className={styles.page}>
      <div className="wrapper">
        <div className="container-fluid">
          <div className="row">
            <div className="col">
              <div className="card">
                <div className="card-body">
                  <div className={styles.cardContent}>
                    <div className={styles.uperContainer}>
                      <div className={styles.imageContainer}>
                        <Image className={styles.profilePicture} width={256} height={256} src="/profile_pic2.jpg" alt="profile picture" />
                      </div>
                      <h5 className="card-title">Jonas Fischer</h5>
                      <div className={styles.factSection}>
                        <div className={styles.factSubContainer}>
                          <Fact factTitle="Developer 7" description="Class & Level" />
                          <Fact factTitle="Catlover/Gamer" description="Background" />
                        </div>
                        <div className={styles.factSubContainer}>
                          <Fact factTitle="Neutral Good" description="Alignment" />
                          <Fact factTitle="Human" description="Race" />
                        </div>
                      </div>
                    </div>
                    <h5>Skills</h5>
                    <div className={styles.progressBarContainer}>
                      <ProgressBar fillingPercentage="75" colorScheme="javascript" title="JavaScript" />
                      <ProgressBar fillingPercentage="85" colorScheme="typescript" title="TypeScript" />
                      <ProgressBar fillingPercentage="85" colorScheme="react" title="React" />
                      <ProgressBar fillingPercentage="75" colorScheme="java" title="Java" />
                      <ProgressBar fillingPercentage="60" colorScheme="springboot" title="Spring Boot" />
                    </div>
                    <h5>Feats</h5>
                    <div className={styles.featsContainer}>
                      <div className="row align-items-center gx-3">
                        <div className="col">
                          <h3><span className="badge bg-primary text-light">Jest</span></h3>
                        </div>
                        <div className="col">
                          <h3><span className="badge bg-primary text-light">Enzyme</span></h3>
                        </div>
                        <div className="col">
                          <h3><span className="badge bg-primary text-light">JUnit</span></h3>
                        </div>
                        <div className="col">
                          <h3><span className="badge bg-primary text-light">Mockito</span></h3>
                        </div>
                        <div className="col">
                          <h3><span className="badge bg-primary text-light">Elasticsearch</span></h3>
                        </div>
                        <div className="col">
                          <h3><span className="badge bg-primary text-light">Git</span></h3>
                        </div>
                        <div className="col">
                          <h3><span className="badge bg-primary text-light">Scrum</span></h3>
                        </div>
                      </div>
                    </div>
                    <h5>Background</h5>
                    <div className={styles.backgroundContainer}>
                      <ul className="list-group list-group-flush">
                        <li className="list-group-item active" aria-current="true">2018 - Today: Software Developer / Immowelt Hamburg GmbH</li>
                        <li className="list-group-item">2015 - 2018: Software Developer Apprenticeship / Immowelt Hamburg GmbH</li>
                        <li className="list-group-item">2013 - 2015: Studies Game Programming / SAE Hamburg (Diploma)</li>
                        <li className="list-group-item">2012 - 2013: Nursery School Teacher Apprenticeship / Fachschule für Sozialpädagogik 2</li>
                        <li className="list-group-item">2012 - 2012: Internships as Nursery School Teacher</li>
                        <li className="list-group-item">2011 - 2012: Call Center Agent / Axel Springer Direkt</li>
                        <li className="list-group-item">2004 - 2011: Studies Media Culture / Universität Hamburg (Bakkalaureus Artium)</li>
                      </ul>
                    </div>
                    <h5>Profiles</h5>
                    <div className="row align-items-center gx-4">
                      <div className="col">
                        <button type="button" className="btn btn-primary" onClick={() => openLink('https://github.com/jfischer84')}>Github</button>
                      </div>
                      <div className="col">
                        <button type="button" className="btn btn-primary" onClick={() => openLink('https://www.xing.com/profile/Jonas_Fischer34/cv')}>Xing</button>
                      </div>
                      <div className="col">
                        <button type="button" className="btn btn-primary" onClick={() => openLink('https://www.linkedin.com/in/jonas-fischer-a5071998/')}>Linkedin</button>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  );
}

export default Index;
