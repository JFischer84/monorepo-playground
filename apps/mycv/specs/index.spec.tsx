import React from 'react';
import {render} from '@testing-library/react';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';

import {Index} from '../pages';

// this test needs to stay out of the pages folder because there is an error with Nextjs
describe('Index', () => {
  it('should render successfully', () => {
    const {baseElement} = render(<Index />);
    expect(baseElement).toBeTruthy();
  });

  it('should render correctly', () => {
    const wrapper = shallow(<Index />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
