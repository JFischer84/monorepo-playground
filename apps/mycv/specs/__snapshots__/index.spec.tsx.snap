// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Index should render correctly 1`] = `
<div
  className="page"
>
  <div
    className="wrapper"
  >
    <div
      className="container-fluid"
    >
      <div
        className="row"
      >
        <div
          className="col"
        >
          <div
            className="navigation"
          >
            <nav>
              <div
                className="nav nav-tabs"
                id="nav-tab"
                role="tablist"
              >
                <button
                  className="nav-item nav-link active"
                  onClick={[Function]}
                  type="button"
                >
                  <div
                    className="navButtonContainer"
                  >
                    <div
                      className="navIconContainer"
                    >
                      <IconEye
                        size={16}
                        stroke={3}
                      />
                    </div>
                    See
                  </div>
                </button>
                <button
                  className="nav-item nav-link "
                  onClick={[Function]}
                  type="button"
                >
                  <div
                    className="navButtonContainer"
                  >
                    <div
                      className="navIconContainer"
                    >
                      <IconCirclePlus
                        size={16}
                        stroke={3}
                      />
                    </div>
                    Create
                  </div>
                </button>
              </div>
            </nav>
            <button
              className="btn btn-outline-danger"
              onClick={[Function]}
              type="button"
            >
              <div
                className="navButtonContainer"
              >
                <div
                  className="navIconContainer"
                >
                  <IconHistory
                    size={16}
                    stroke={3}
                  />
                </div>
                Reset
              </div>
            </button>
          </div>
          <div
            className="card"
          >
            <div
              className="card-body"
            >
              <Index
                cvData={
                  Object {
                    "alignment": "Neutral Good",
                    "background": "Catlover/Gamer",
                    "classAndLevel": "Developer 7",
                    "name": "Jonas Fischer",
                    "species": "Human",
                  }
                }
                feats={
                  Array [
                    "Jest",
                    "Enzyme",
                    "SCSS",
                    "ESLint",
                    "Stylelint",
                    "ES6",
                    "Next.js",
                    "Webpack",
                    "JUnit",
                    "Mockito",
                    "Elasticsearch",
                    "Git",
                    "Scrum",
                  ]
                }
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
